### DISCUSSION TITLE
How to use context for translation

### DISCUSSION THREAD

**Discussion by radoam1** [i18next/next-i18next#2133] [Q&A]
Hi, I am new in next js world and I try using context for my translations I mean I have a translation file like bellow

common.json
`
{
  "fruit_apple": "Apple",
  "fruit_banana": "Banana"
}
`

and I try to pass `apple` context for translation and it don't work

`
t("fruit", {
  context: "apple"
})
`

I want to achieve that I can manipulate context and translation is changing depend on context it is even possible?

**Comment by aaimio** [comment]
You could do something like this: `t("Is your favourite fruit {{ fruit }}?", { fruit: "apple" })`

See https://www.i18next.com/translation-function/interpolation

**Reply by radoam1** [reply]
Yea I know but I want to achieve this effect:

[https://www.i18next.com/translation-function/context](https://www.i18next.com/translation-function/context)

**Reply by adrai** [reply]
should work: https://youtu.be/SA_9i4TtxLQ?t=653

**Reply by radoam1** [reply]
So I watched this video and I don't know in my case it didn't work bellow my configuration:

i18n.js
```js
module.exports = {
  i18n: {
    defaultLocale: 'pl',
    locales: ['pl'],
    defaultLanguage: 'pl',
    fallbackLng: ['pl'],
  },
};
```
next-i18next.config.ts
```js
import ICU from 'i18next-icu';

const { i18n } = require('./i18n');

const config = {
  i18n,
  use: [ICU],
  serializeConfig: false,
};

export default config;
```

of course I import `i18n` and applay to `next.config.js`

this is strange because everything else works fine but if I want to pass variable like you write before I need to use single curly bracket like `t("Is your favourite fruit { fruit }?", { fruit: "apple" })`, its strange for me because I have a default configuration

**Reply by adrai** [reply]
Because you're using ICU... that's not the i18next format.
ICU is not aware of the context feature.

The ICU syntax is completely different.

Have a look at: https://locize.com/blog/i18n-formats-javascript/#icu

**Reply by radoam1** [reply]
Ohh thanks for it I though that it's not a problem thanks again for help
